src/data/GraphOnChipContext.tsx

```diff
@@ -107,19 +86,41 @@ const GraphOnChipProvider: React.FC<{ children: ReactNode }> = ({ children }) =>
             })
             .flat();
         setState({
-            visitedGraphsHistory: [],
-            currentGraphIndex: 0,
             graphOnChipList: Object.fromEntries(newChips.map((chip, index) => [graphs[index].name, chip])),
             graphs: new Map(graphs.map((graph) => [graph.name, graph])),
             operands: new Map(operands.map((edge) => [edge.name, edge])),
         });
     }, []);
 
     const getGraphOnChip = useCallback(
-        (graphName: string) => {
-            return state.graphOnChipList[graphName];
+        (temporalEpoch: number, chipId?: number) => {
+            const graphs: { graph: GraphOnChip; relationship: GraphRelationship }[] = [];
+
+            [...state.graphs.values()].forEach((graphRelationship) => {
+                const isSameTemporalEpoch = graphRelationship.temporalEpoch === temporalEpoch;
```

Comment:
nit: i find those variable declaration to be overly verbose, would suggest inlining and if/else

Reply:
Agree, it is super verbose, but IMO this is more readable:
```typescript
const isSameTemporalEpoch = graphRelationship.temporalEpoch === temporalEpoch;
const hasChipId = chipId !== null && chipId !== undefined;
const isSameChipId = graphRelationship.chipId === chipId;

if (isSameTemporalEpoch) {
    if (!hasChipId) {
        graphs.push({
            graph: state.graphOnChipList[graphRelationship.name],
            relationship: graphRelationship,
        });
    }

    if (hasChipId && isSameChipId) {
        graphs.push({
            graph: state.graphOnChipList[graphRelationship.name],
            relationship: graphRelationship,
        });
    }
}
```

Than this:
```typescript
if (graphRelationship.temporalEpoch === temporalEpoch) {
    if (chipId === null || chipId === undefined) {
        graphs.push({
            graph: state.graphOnChipList[graphRelationship.name],
            relationship: graphRelationship,
        });
    }

    if ((chipId !== null && chipId !== undefined) && graphRelationship.chipId === chipId) {
        graphs.push({
            graph: state.graphOnChipList[graphRelationship.name],
            relationship: graphRelationship,
        });
    }
}
```

Reply:
how about this?
```
if (graphRelationship.temporalEpoch === temporalEpoch) {
                    if (chipId === null || chipId === undefined) {
                        graphs.push({
                            graph: state.graphOnChipList[graphRelationship.name],
                            relationship: graphRelationship,
                        });
                    } else if (graphRelationship.chipId === chipId) {
                        graphs.push({
                            graph: state.graphOnChipList[graphRelationship.name],
                            relationship: graphRelationship,
                        });
                    }
                }
```